src/components/global.navigation.ts

```diff
@@ -1,37 +1,62 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnDestroy, OnInit } from '@angular/core';
 import { Services } from '../services';
 import { TraversingComponent } from '../traversing';
 import { NavLink } from '../interfaces';
 import { Target } from 'angular-traversal';
+import { Subscription } from 'rxjs/Subscription';
+import { ResourceService } from '../resource.service';
+import { Observable } from 'rxjs/Observable';
 
 
 @Component({
   selector: 'plone-global-navigation',
-  template: `<ul>
-  <li *ngFor="let link of links" [ngClass]="{'active': link.active}">
-    <a [traverseTo]="link.path">{{ link.title }}</a>
-  </li>
-</ul>`
+  template: `
+    <ul>
+      <li *ngFor="let link of links" [ngClass]="{'active': link.active}">
+        <a [traverseTo]="link.path">{{ link.title }}</a>
+      </li>
+    </ul>`
 })
-export class GlobalNavigation extends TraversingComponent implements OnInit {
+export class GlobalNavigation extends TraversingComponent implements OnInit, OnDestroy {
 
+  target: Target | null = null;
   links: NavLink[] = [];
+  navigationRefresh: Subscription;
 
-  constructor(
-    public services: Services,
-  ) {
+  constructor(public services: Services,) {
     super(services);
   }
 
   ngOnInit() {
     super.ngOnInit();
-    this.services.resource.navigation()
+    const component = this;
```

Comment:
is it useful?
(to me, as far as we are in arrow functions, this is the right this)

Reply:
I was wondering the same thing